spring-cloud-starter-stream-sink-log/src/main/java/org/springframework/cloud/stream/app/log/sink/LogSinkConfiguration.java

```diff
@@ -40,7 +47,20 @@
 	@Bean
 	@ServiceActivator(inputChannel = Sink.INPUT)
 	public LoggingHandler logSinkHandler() {
-		LoggingHandler loggingHandler = new LoggingHandler(this.properties.getLevel().name());
+		LoggingHandler loggingHandler = new LoggingHandler(this.properties.getLevel().name()) {
+
+			@Override
+			protected void handleMessageInternal(Message<?> message) throws Exception {
+				String contentType = message.getHeaders().containsKey(MessageHeaders.CONTENT_TYPE)
+						? message.getHeaders().get(MessageHeaders.CONTENT_TYPE).toString()
+						: BindingProperties.DEFAULT_CONTENT_TYPE.toString();
```

Comment:
Soby, I don't think that is correct. We need to get the actual binding content-type in the event it was set. Obviously it would be the edge case, but still from the standpoint of correctness that is what we should do and fail if it is not `application/json`

Reply:
@sobychacko you are so, so, so, so, so right!!!! I've completely forgot about the fact that interceptor will inject the appropriate CT as header ALWAYS. So technically the who IF statement is only for extra safety since the else will never get triggered under normal conditions.
My bad!

Reply:
Wait.
`DEFAULT_CONTENT_TYPE`? 
You always are going to convert incoming `byte[]` to string, unless I'll specify an explicit non-`text`, non-`json`.

No? Am I missing anything else?

Reply:
@artembilan yes you are missing "something" ;) Read my previous comment. The IF/ELSE is primarily for correctness. CT will always be there under normal conditions, so the ELSE will never trigger, yet a reasonable default.

Reply:
Yeah... Never mind.

Let me ask a different way.

What is that `CT` when:

>  that interceptor will inject the appropriate CT as header ALWAYS

?

I mean the data on the wire is `byte[]` and without `CT`. According your explanation you are going to inject some `CT` in the interceptor anyway before pushing message to the consumer. Therefore what is that default `CT` do you use?

Thanks

Reply:
No, the wire-format implies the payload only. So you get `Message<byte[]>` and the header passed by the producer (if set) is there and if not the binding and/or default will be injected

Reply:
the actual wire bytes are converted to Message by corresponding SI adapters (i.e., Rabbit, Kafka)

`byte[] -> InboundChannelAdapter -> Message<byte[]> -> application-handler`